Userland/Libraries/LibGL/Tex/TextureUnit.cpp

```diff
@@ -14,20 +14,23 @@ void TextureUnit::bind_texture_to_target(GLenum texture_target, const RefPtr<Tex
     switch (texture_target) {
     case GL_TEXTURE_2D:
         m_texture_target_2d = static_ptr_cast<Texture2D>(texture);
-        m_currently_bound_texture = texture;
         m_currently_bound_target = GL_TEXTURE_2D;
+        m_texture_2d_enabled = true;
```

Comment:
Are you sure about that? I don't think binding a texture automatically enables texturing.

Reply:
I believe you are right - I cannot find anything that supports this behavior. I guess I assumed it based on the way 3DFileViewer was implemented. Fixed!